third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -311,128 +311,101 @@ Value llGetPid(Location loc, RewriterBase &rewriter, ModuleOp moduleOp,
 Value llLoad(RewriterBase &rewriter, Location loc, Value ptr, Type elemTy,
              Value pred, Value falseVal, triton::CacheModifier cm,
              bool forceNoAliasAsyncLoads) {
-  auto b = TritonLLVMOpBuilder(loc, rewriter);
+  bool isNonMasked = false;
+
+  if (auto constOp = pred.getDefiningOp<LLVM::ConstantOp>()) {
```

Comment:
Can we fold this case into the lowering of `MaskedLoadOp`? So that for example we can reuse the following volatile and nontemporal mapping. Similarly for the store op.